data_steward/retraction/retract_data_bq.py

```diff
@@ -439,34 +354,54 @@ def skip_retraction(client, dataset_id, table_id, retraction_type) -> bool:
         elif table_id == PERSON:
             LOGGER.info(msg_only_rdr)
             return True
-        elif ru.is_deid_dataset(dataset_id) and not client.table_exists(
+        elif is_deid_dataset(dataset_id) and not client.table_exists(
                 f'{table_id}_ext', dataset_id):
             LOGGER.info(msg_no_mapping_table)
             return True
-        elif ru.is_combined_dataset(dataset_id) and not client.table_exists(
+        elif is_combined_dataset(dataset_id) and not client.table_exists(
                 mapping_table_for(table_id), dataset_id):
             LOGGER.info(msg_no_mapping_table)
             return True
 
     return False
 
 
-def get_only_ehr_condition(client: BigQueryClient, dataset_id, table_id) -> str:
+def skip_dataset_retraction(dataset_id, hpo_id, retraction_type) -> bool:
     """
-    Generates WHERE clause for only_ehr retraction.
-    :param client: BigQuery client
+    Some datasets do not need retraction depending on how we want to retract. 
+    This function returns True if the dataset does not need retraction.
     :param dataset_id: dataset to run retraction for
-    :param table_id: table to run retraction for
-    :return: WHERE clause in str format to be added when only_ehr
+    :param hpo_id: hpo_id of the site to retract from
+    :param retraction_type: string indicating whether all data needs to be removed including RDR,
+        or if RDR data needs to be kept intact. Can take the values 'rdr_and_ehr' or 'only_ehr'
+    :return: True if the dataset should be skipped. False if we need to retract the dataset.
     """
-    if table_id == DEATH:
-        return ''
+    msg_skip = f'Skipping retraction for the dataset {dataset_id}, '
+    msg_missing_hpo_id = (
+        'because hpo_id is not specified. This is an EHR dataset and hpo_id '
+        'must be defined when retracting from an EHR dataset.')
```

Comment:
I'm not convinced this is completely accurate.  If we're retracting EHR data because the person never gave consent, then we should be retracting all EHR data.  If we're retracting data related to one site because of the site's process, then we would want to remove data from that site only.

Reply:
Thanks, @lrwb-aou, this is another point I need your advice on.
I agree with your point. The records must be deleted from all EHR data in such a case. The reason I kept the original logic that `hpo_id` is mandatory for EHR retraction is that the retraction for an EHR dataset takes long if we go through all the tables in it because it has `xyz_observation`, `xyz_measurement`, and so on, for every EHR site. 

I think I can update the logic like the following. Does this look good to you?
- If retracting an EHR dataset and `hpo_id` is specified, only tables from the HPO site are retracted.
- If retracting an EHR dataset and `hpo_id` is NOT specified, all tables in the dataset are retracted. It will take longer to finish retraction but it is not critical.

Reply:
@hiro-mishima I agree with your bullet points.  So, we might want to alter this message and raise it as at least a warning that the user must then agree to.  This seems like the safest course of action.  What do you think?

Reply:
@lrwb-aou Sounds good! Let me update the scripts accordingly.